src/HIE/Bios/Cradle/Cabal.hs

```diff
@@ -332,8 +378,13 @@ processCabalWrapperArgs args =
 -- ----------------------------------------------------------------------------
 
 cabalLoadFilesBefore315 :: LogAction IO (WithSeverity Log) -> ProgramVersions -> CradleProjectConfig -> [Char] -> [String] -> CradleLoadResultT IO CreateProcess
-cabalLoadFilesBefore315 l progVersions projectFile workDir args = do
+cabalLoadFilesBefore315 l progVersions projectFile workDir args' = do
   let cabalCommand = "v2-repl"
+  cabal_version <- liftIO $ runCachedIO $ cabalVersion progVersions
+
+  let args = case cabal_version of
+        Just v | v < makeVersion [3,15] -> "--keep-temp-files" : args'
+        _ -> args'
```

Comment:
For cabal version >= 3.15, we use cabalLoadFilesWithRepl with `--keep-temp-files` as one of the arg
For cabal version < 3.15 , we use cabalLoadFilesBefore315, where we do not have `--keep-temp-files`.

In windows, run without `--keep-temp-files`, `multi-out` file is being cleanup too early causing the problem.

Here I added back `--keep-temp-files` to keep `multi-out` around for cabalLoadFilesBefore315. Do you think it is a good workaround? @fendor

Reply:
Yes, unfortunately somehow v2-repl does not take --keep-temp-files as argument after 3.15

Reply:
Probably in conjunction with `--with-repl`, the flag isn't supported, but it looks like an oversight, tbh.